linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -1046,20 +1122,66 @@ object Analyzer {
   def computeReachability(config: CommonPhaseConfig,
       symbolRequirements: SymbolRequirement,
       allowAddingSyntheticMethods: Boolean,
-      inputProvider: InputProvider): Analysis = {
+      inputProvider: InputProvider)(implicit ex: ExecutionContext): Future[Analysis] = {
     val analyzer = new Analyzer(config, symbolRequirements,
-        allowAddingSyntheticMethods, inputProvider)
-    analyzer.computeReachability()
-    analyzer
+        allowAddingSyntheticMethods, inputProvider, ex)
+    analyzer.computeReachability().map(_ => analyzer)
   }
 
   trait InputProvider {
-    def classesWithEntryPoints(): TraversableOnce[String]
+    def classesWithEntryPoints()(implicit ex: ExecutionContext): Future[TraversableOnce[String]]
 
-    def loadInfo(encodedName: String): Option[Infos.ClassInfo]
+    def loadInfo(encodedName: String)(implicit ex: ExecutionContext): Option[Future[Infos.ClassInfo]]
```

Comment:
Shouldn't this be a `Future[Option[...]]` instead?

Reply:
Well, in the current implementation we do know whether a file exists or not without an async operation (and this will stay that way). So we can make it a `Option[Future[...]]`.